src/ui/components/Library/Team/View/Tests/hooks/useSyncTestIdToUrl.ts

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from "next/router";
+import { Dispatch, SetStateAction, useEffect, useRef } from "react";
+
+import { parseQueryParams } from "ui/components/Library/Team/utils";
+
+export function useSyncTestIdToUrl(
+  teamId: string,
+  testId: string | null,
+  setTestId: Dispatch<SetStateAction<string | null>>
+) {
+  const router = useRouter();
+
+  const prevTestIdRef = useRef(testId);
+
+  // Sync component state changes to URL
+  useEffect(() => {
+    const prevTestId = prevTestIdRef.current;
+    if (testId && testId !== prevTestId) {
+      const pathname = `/team/${teamId}/tests/${testId}`;
+      if (prevTestId === null) {
+        router.replace({ pathname, query: router.query });
+      } else if (prevTestIdRef.current !== testId) {
+        router.push({ pathname, query: router.query });
+      }
+    }
+
+    prevTestIdRef.current = testId;
+  }, [router, teamId, testId]);
+
+  // Sync URL changes to component state
+  useEffect(() => {
+    const testIdFromState = prevTestIdRef.current;
+    const { testOrTestRunId: testIdFromUrl } = parseQueryParams(router.query);
+    if (testIdFromUrl) {
+      if (testIdFromState !== testIdFromUrl) {
+        setTestId(testIdFromUrl);
+      }
+    }
+  }, [router.query, setTestId]);
```

Comment:
I think the right way would be to have this state only in the URL without duplicating it in `TestContext`. In other words `TestContext` would need to be changed so that its `testId` comes from the URL and its `setTestId()` writes directly to the URL.

Reply:
Agreed. This will be removed by SCS-1793

Reply:
I didn't realize there were pending bug-fixes for merge into test-suites. Seems like we're reviewing this prematurely.